### PR TITLE
fix(channels): restore liveness on stuck requests

### LINKED ISSUE #1675: [Bug]: Message queue blocked when _process_batch hangs due to missing timeout protection
## Environment
- CoPaw: 0.0.7
- OS: Windows 11
- Python: 3.x

## Problem Description

When `_process_batch` is executing and the LLM API call hangs (e.g., due to network issues, API timeout, etc.), the entire message queue becomes blocked:

1. `_process_batch` starts executing → `_in_progress = True`
2. LLM API call hangs (no timeout protection)
3. `_in_progress` is never cleared
4. New messages enter `pending` queue and wait
5. All message processing is blocked

### Key Log
\copaw\app\channels\manager.py:230 | 2026-03-17 19:44:08 | manager _enqueue_one dingtalk: key=ZFsk+2A= in_progress=True payload_has_sw=True -> pending copaw

## Root Cause

The `_process_batch` function lacks timeout protection. When the LLM API call hangs, `_in_progress` remains `True` indefinitely, blocking the entire message queue.

## Expected Behavior

- `_process_batch` should have a timeout mechanism
- When timeout occurs, `_in_progress` should be properly cleaned up
- New messages should not be blocked by a hanging previous request

## Suggested Solution

Add timeout protection to `_process_batch`:

```python
async def _process_batch(ch: BaseChannel, batch: List[Any]) -> None:
    try:
        # Add timeout protection (e.g., 5 minutes)
        await asyncio.wait_for(
            _process_batch_internal(ch, batch),
            timeout=300
        )
    except asyncio.TimeoutError:
        logger.error("_process_batch timeout for channel=%s", ch.channel)
        # Ensure _in_progress is cleaned up
    except Exception as e:
        logger.error("_process_batch error: %s", e)
        # Ensure _in_progress is cleaned up
```

## Workaround
Currently, the only workaround is to restart the copaw process:

```python
Stop current copaw
taskkill /F /PID <pid>

Restart copaw
copaw run
```

### PR DESCRIPTION
**Pull request by fancyboi999**
## Summary
- restore standard asyncio cancellation propagation in `AgentRunner` so upstream timeouts behave predictably
- add explicit runtime LLM client timeouts for OpenAI, Anthropic, Gemini, and Ollama-backed chat models
- add a channel-level watchdog that releases stuck per-session work and sends a timeout error instead of leaving the queue blocked
- cover the regression with runner cancellation, channel timeout, and provider timeout injection tests

## Root Cause
Issue #1675 identifies the symptom correctly but not the whole failure chain. The real problem had three layers:
1. `ChannelManager` held `_in_progress` until batch processing returned, with no watchdog.
2. Runtime chat-model instances were created without a CoPaw-owned timeout policy, so the system relied on SDK defaults.
3. `AgentRunner` converted `CancelledError` into `RuntimeError`, which breaks `asyncio.wait_for` semantics and makes higher-level timeout handling brittle.

## Solution
- Preserve `CancelledError` after cleanup in `AgentRunner`.
- Introduce `COPAW_LLM_REQUEST_TIMEOUT_SECONDS` and `COPAW_CHANNEL_PROCESS_TIMEOUT_SECONDS`.
- Inject runtime timeout settings when constructing provider-backed chat models.
- Wrap channel batch execution in a watchdog and send a terminal timeout error on expiry.

## Validation
- `python -m pytest tests/unit/runner/test_runner_cancellation.py tests/unit/channels/test_manager_timeout.py tests/unit/providers/test_openai_provider.py tests/unit/providers/test_anthropic_provider.py tests/unit/providers/test_gemini_provider.py tests/unit/providers/test_ollama_provider.py -q`
- `python -m pytest tests/unit/workspace/test_workspace.py -q`

Closes #1675

### CODE REVIEW

File: src/copaw/app/channels/manager.py [hunk 1]
```diff
@@ -78,17 +83,67 @@ async def _process_batch(ch: BaseChannel, batch: List[Any]) -> None:
                 "manager _process_batch dingtalk merged: has_sw=%s",
                 bool(merged.get("session_webhook")),
             )
-        await ch._consume_one_request(merged)
-    elif len(batch) > 1:
+        return merged, False
```
[the remark this answers is not in the document]
Comment (fancyboi999) [reply src/copaw/app/channels/manager.py]: Handled in c278412.

I added a guard in `_resolve_batch_target()` so the native-merge branch only returns the merged payload when `merge_native_items()` returns a real value. When it returns `None`, the code now falls back to `(batch[0], True)` instead of forwarding `None` into `_consume_one_request()`.

I also added a regression test covering the `merge_native_items() -> None` path in `tests/unit/channels/test_manager_timeout.py`.
